cmd/connectivity.go

```diff
@@ -0,0 +1,142 @@
+package cmd
+
+import (
+	"context"
+	"encoding/base64"
+	"errors"
+	"fmt"
+	"os"
+	"time"
+
+	grpc_middleware "github.com/grpc-ecosystem/go-grpc-middleware"
+	grpc_retry "github.com/grpc-ecosystem/go-grpc-middleware/retry"
+	grpc_prometheus "github.com/grpc-ecosystem/go-grpc-prometheus"
+	"github.com/mattn/go-isatty"
+	"github.com/odpf/salt/log"
+	"github.com/spf13/afero"
+	"go.opentelemetry.io/contrib/instrumentation/google.golang.org/grpc/otelgrpc"
+	"google.golang.org/grpc"
+
+	"github.com/odpf/optimus/config"
+)
+
+const (
+	GRPCMaxClientSendSize      = 64 << 20 // 64MB
+	GRPCMaxClientRecvSize      = 64 << 20 // 64MB
+	GRPCMaxRetry          uint = 3
+
+	OptimusDialTimeout = time.Second * 2
+	BackoffDuration    = 100 * time.Millisecond
+)
+
+const (
+	defaultProjectName = "sample_project"
+	defaultHost        = "localhost:9100"
+)
+
+type BearerAuthentication struct {
+	Token string
+}
+
+func (a *BearerAuthentication) GetRequestMetadata(context.Context, ...string) (map[string]string, error) {
+	return map[string]string{
+		"Authorization": fmt.Sprintf("Bearer %s", a.Token),
+	}, nil
+}
+
+func (a *BearerAuthentication) RequireTransportSecurity() bool {
+	return false
+}
+
+type BasicAuthentication struct {
+	Token string
+}
+
+func (a *BasicAuthentication) GetRequestMetadata(context.Context, ...string) (map[string]string, error) {
+	return map[string]string{
+		"Authorization": fmt.Sprintf("Basic %s", a.Token),
+	}, nil
+}
+
+func (a *BasicAuthentication) RequireTransportSecurity() bool {
+	return false
+}
+
+func isTerminal(f *os.File) bool {
+	return isatty.IsTerminal(f.Fd()) || isatty.IsCygwinTerminal(f.Fd())
+}
+
+func createDataStoreSpecFs(namespace *config.Namespace) map[string]afero.Fs {
+	dtSpec := make(map[string]afero.Fs)
+	for _, dsConfig := range namespace.Datastore {
+		dtSpec[dsConfig.Type] = afero.NewBasePathFs(afero.NewOsFs(), dsConfig.Path)
+	}
+	return dtSpec
+}
+
+func initClientConnection(l log.Logger, serverHost string, requestTimeout time.Duration) (
+	requestCtx context.Context,
+	connection *grpc.ClientConn,
+	closeConnection func(),
+	err error,
+) {
+	connection, err = createConnection(serverHost)
+	if err != nil {
+		return
+	}
+	reqCtx, reqCancel := context.WithTimeout(context.Background(), requestTimeout)
+
+	requestCtx = reqCtx
+	closeConnection = func() {
+		l.Info("Closing client connection")
```

Comment:
Is this for testing or do we need this info log ?

Reply:
oh right, missed that one. should've removed it

Reply:
it is addressed